lib/screens/wallet/wallet_screen.dart

```diff
@@ -301,7 +301,7 @@ class _SignScreenState extends State<SignScreen> with WidgetsBindingObserver {
     final analyticManager = Provider.of<AnalyticManager>(context, listen: false);
     final chainLoader = Provider.of<ChainLoader>(context, listen: false);
     _signRequestsSubscription?.pause();
-    Chain? chain = requst.chainId == null ? null : chainLoader.getChainInfo(requst.chainId!);
```

Comment:
@noahsilencelab probably there is no reason for us to block UI thread completely and wait for parsing result. 
What do you think about using Future<Chain> in `SingRequestViewModel`, and utilize `FutureBuilder` in UI?

Reply:
@Granfalloner it is a good idea. The implementation: [#ffeabad](https://github.com/silence-laboratories/silent-shard-mobile/pull/19/commits/ffeabadeeec87b7c40d7efc0cf23769c8b8257d8)